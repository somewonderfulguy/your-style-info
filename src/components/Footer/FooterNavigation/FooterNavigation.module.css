.link {
  color: #fff;
}

.active {
  color: lightgray;
  font-weight: 600;
}