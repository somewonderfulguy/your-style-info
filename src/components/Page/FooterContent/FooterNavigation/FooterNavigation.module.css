.list {
  list-style-type: none;
  display: grid;
  grid-template-rows: repeat(4, min-content);
  grid-auto-flow: column;
  grid-auto-columns: 250px;
  gap: 10px;
}

@media (max-width: 425px) {
  .list {
    grid-template-rows: initial;
    grid-auto-flow: initial;
    grid-auto-columns: auto;
  }
}

.link {
  color: #919191;
}

.link:hover {
  color: lightgray;
}

.active {
  color: lightgray;
  font-weight: 600;
}

.inactive {
  color: #4a4a4a;
}
