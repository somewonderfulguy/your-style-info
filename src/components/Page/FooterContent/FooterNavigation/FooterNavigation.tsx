import LinkExtended from '~components/LinkExtended'
import { PRIME_ROUTES } from '~constants/index'
import { useLocalization } from '~contexts/localizationContext'

import styles from './FooterNavigation.module.css'

const FooterNavigation = () => {
  const [{ locale }] = useLocalization()

  return (
    <nav>
      <ul className={styles.list}>
        {Object.entries(PRIME_ROUTES) &&
          Object.entries(PRIME_ROUTES).map(([path, { name, inactive }]) => (
            <li key={path}>
              {inactive ? (
                <span className={styles.inactive} aria-disabled>
                  {name}
                </span>
              ) : (
                <LinkExtended
                  to={`/${locale}${path}`}
                  className={styles.link}
                  activeClassName={styles.active}
                >
                  {name}
                </LinkExtended>
              )}
            </li>
          ))}
      </ul>
    </nav>
  )
}

export default FooterNavigation
