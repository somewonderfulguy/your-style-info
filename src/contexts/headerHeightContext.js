import React, {createContext, useContext, useMemo, useState} from 'react'

export const ERROR_HEADER_HEIGHT = 'useHeaderHeight must be used within a HeaderHeightProvider'

const HeaderHeightContext = createContext()

const useHeaderHeight = () => {
  const context = useContext(HeaderHeightContext)
  if(!context) {
    throw new Error(ERROR_HEADER_HEIGHT)
  }
  return context
}

const HeaderHeightProvider = props => {
  const [headerHeight, setHeaderHeight] = useState(0)
  const value = useMemo(() => ({headerHeight, setHeaderHeight}), [headerHeight])
  return <HeaderHeightContext.Provider value={value} {...props} />
}

export {HeaderHeightProvider, useHeaderHeight}