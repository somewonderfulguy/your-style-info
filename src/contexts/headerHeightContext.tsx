import {
  createContext,
  Dispatch,
  ReactNode,
  SetStateAction,
  useContext,
  useState
} from 'react'

const HeaderHeightStateContext = createContext<number | undefined>(undefined)
HeaderHeightStateContext.displayName = 'HeaderHeightStateContext'

const HeaderHeightDispatchContext = createContext<
  Dispatch<SetStateAction<number>> | undefined
>(undefined)
HeaderHeightDispatchContext.displayName = 'HeaderHeightDispatchContext'

const HeaderHeightProvider = ({ children }: { children: ReactNode }) => {
  const [headerHeight, setHeaderHeight] = useState(0)
  return (
    <HeaderHeightStateContext.Provider value={headerHeight}>
      <HeaderHeightDispatchContext.Provider value={setHeaderHeight}>
        {children}
      </HeaderHeightDispatchContext.Provider>
    </HeaderHeightStateContext.Provider>
  )
}

const useHeaderHeightState = () => {
  const context = useContext(HeaderHeightStateContext)
  if (context === undefined) {
    throw new Error(
      'useHeaderHeightState must be used within a HeaderHeightProvider'
    )
  }
  return context
}

const useHeaderHeightDispatch = () => {
  const context = useContext(HeaderHeightDispatchContext)
  if (context === undefined) {
    throw new Error(
      'useHeaderHeightDispatch must be used within a HeaderHeightProvider'
    )
  }
  return context
}

export { HeaderHeightProvider, useHeaderHeightState, useHeaderHeightDispatch }
