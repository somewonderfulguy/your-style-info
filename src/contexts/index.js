export * from './headerHeightContext'
export * from './loadingContext'
export * from './localisationContext'
export * from './screenDimensionsContext'
export * from './themeContext'