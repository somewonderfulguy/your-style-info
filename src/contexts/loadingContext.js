import React, {createContext, useContext, useMemo, useState} from 'react'

export const ERROR_LOADING = 'useLoading must be used within a LoadingProvider'

const LoadingContext = createContext()
LoadingContext.displayName = 'LoadingContext'

const useLoading = () => {
  const context = useContext(LoadingContext)
  if(!context) {
    throw new Error(ERROR_LOADING)
  }
  const [isLoading, setLoading] = context

  return {isLoading, setLoading}
}

const LoadingProvider = props => {
  const [isLoading, setLoading] = useState(false)

  const value = useMemo(() => [isLoading, setLoading], [isLoading])
  return <LoadingContext.Provider value={value} {...props} />
}

export {LoadingProvider, useLoading}