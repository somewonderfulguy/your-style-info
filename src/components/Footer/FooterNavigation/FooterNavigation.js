import React from 'react'

import LinkExtended from 'components/LinkExtended'
import {PRIME_ROUTES} from 'constants/index'
import {useLocalisation} from 'contexts'
import styles from './FooterNavigation.module.css'

const FooterNavigation = () => {
  const {locale} = useLocalisation()

  return (
    <nav>
      <ul className={styles.list}>
        {Object.entries(PRIME_ROUTES) && Object.entries(PRIME_ROUTES).map(([path, {name, inactive}]) => (
          <li key={path}>
            {inactive ? (
              <span className={styles.inactive} aria-disabled>{name}</span>
            ) : (
              <LinkExtended to={`/${locale}${path}`} className={styles.link} activeClassName={styles.active}>
                {name}
              </LinkExtended>
            )}
          </li>
        ))}
      </ul>
    </nav>
  )
}

export default FooterNavigation