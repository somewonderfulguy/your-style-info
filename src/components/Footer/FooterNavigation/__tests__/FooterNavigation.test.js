import React from 'react'
import {render, screen} from '@testing-library/react'

import FooterNavigation from '..'
import {LocalisationProvider} from 'contexts'

jest.mock('constants/primeRoutes')
jest.mock('components/LinkExtended', () => ({to, children}) => <a href={to}>{children}</a>)

const setup = () => {
  const utils = render(<FooterNavigation />, {wrapper: LocalisationProvider})
  return utils
}

test('should match snapshot', () => {
  const {asFragment} = setup()
  expect(asFragment()).toMatchSnapshot()
})

test('should render menu items as links and inactive items as usual text', () => {
  setup()

  const listItems = screen.getAllByRole('listitem')
  const links = screen.getAllByRole('link')
  const inactiveItems = screen.getByRole('list').querySelectorAll('*[aria-disabled="true"]')

  expect(listItems).toHaveLength(3)
  expect(links).toHaveLength(2)
  expect(inactiveItems).toHaveLength(1)
})