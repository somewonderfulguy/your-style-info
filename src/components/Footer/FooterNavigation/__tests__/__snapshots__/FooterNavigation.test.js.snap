// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="list"
    >
      <li>
        <a
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/guides-topics"
        >
          Generic topics & guides
        </a>
      </li>
      <li>
        <span
          aria-disabled="true"
          class="inactive"
        >
          Mixed
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
